src/mod/gcewing/sg/features/gdo/client/gui/GateControlComponent.java

```diff
@@ -0,0 +1,168 @@
+                .text(this::address)
+                .textColor(0x5555FF)
+                .scale(1.3F)
+                .build();
+
+        UIButton irisControl = UIButton.builder()
+                .parent(this)
+                .leftAligned()
+                .below(address, 5)
+                .size(70, 20)
+                .text(this::irisControlText)
+                .enabled(this::hasIrisAccess)
+                .onClick(this::openCloseIris)
+                .build();
+
+        if (local) {
+            UIButton.builder()
+                    .parent(this)
+                    .rightOf(irisControl, 5)
+                    .below(address, 5)
+                    .size(70, 20)
+                    .text("sgcraft.gui.button.disconnect")
+                    .enabled(this::isGateConnected)
+                    .onClick(this::disconnectGate)
+                    .build();
+        }
+
+    }
+
+    public void setGate(SGBaseTE gate) {
+        this.gate = gate;
+    }
+
+    public void setIrisAccess(boolean irisAccess) {
+        this.irisAccess = irisAccess;
+    }
+
+    public GuiIcon gateStatus() {
+        if (gate == null) {
+            return GuiIcon.NONE;
+        }
+
+        return gateTextures.computeIfAbsent(gateStatusLabel(), s -> GuiIcon.full(new GuiTexture(new ResourceLocation("sgcraft:" + s))));
+    }
+
+    private String gateStatusLabel() {
+        String status = "textures/";
+        status += gate.gateType == 2 ? "pegasus_" : "milkyway_";
+        status += isGateConnected() ? "connected_" : "disconnected_";
+        status += gate.hasIrisUpgrade ? "iris_" + (gate.irisIsClosed() ? "closed" : "open") : "no_iris";
+        status += ".png";
+        return status;
+    }
+
+    private String address() {
+        return gate != null ? SGAddressing.formatAddress(gate.homeAddress, "-", "-") : "";
+    }
+
+    private String irisControlText() {
+        if (gate == null) {
+            return "";
+        }
+
+        return gate.hasIrisUpgrade && gate.irisIsClosed() ? "sgcraft.gui.button.openIris" : "sgcraft" + ".gui.button.closeIris";
+    }
+
+    private boolean hasIrisAccess() {
+        return irisAccess && gate != null && gate.hasIrisUpgrade;
+    }
+
+    private void openCloseIris() {
+        if (gate == null || !gate.hasIrisUpgrade) {
+            return;
+        }
+        GdoNetworkHandler.sendGdoInputToServer(gate, gate.irisIsClosed() ? 1 : 2);
+    }
+
+    private boolean isGateConnected() {
+        return gate != null && gate.isConnected() && gate.state != SGState.SyncAwait;
+    }
+
+    private void disconnectGate() {
+        GdoNetworkHandler.sendGdoInputToServer(gate, 3);
+    }
+
+    public static GateControlComponentBuilder builder(SGBaseTE gate, boolean local) {
+        return new GateControlComponentBuilder(gate, local);
+    }
+
+    public static class GateControlComponentBuilder extends UIContainerBuilderG<GateControlComponentBuilder, GateControlComponent> {
```

Comment:
This makes the class lines needlessly long for these.

```java
GateControlComponent.GateControlComponentBuilder builder...
```

These should all be called Builder, so it becomes clean:

```java
GateControlComponent.Builder builder = ...
```